packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -539,12 +539,12 @@ Future<void> _buildGradleProjectV2(
 
         final String name = newFile.name;
         if (name.contains('_snapshot_') || name.endsWith('.so')) {
-          final List<int> diff = bsdiff(oldFile.content, newFile.content);
-          final int ratio = 100 * diff.length ~/ newFile.content.length;
+          final List<int> diff = bsdiff(oldFile.content as List<int>, newFile.content as List<int>);
+          final int ratio = 100 * diff.length ~/ (newFile.content.length as int);
```

Comment:
is length not statically an int?

Reply:
Like above, all of the file/devfs/process APIs use dynamic

Reply:
sure but the most type-safe solution here would be to cast the `content`, not the `length`.